Tests failing due to depreciated Gemini model

Looks like we have a test referencing gemini-2.5-pro which was depreciated today.

See logs:
https://github.com/freelawproject/courtlistener/actions/runs/27708898095/job/81964816315?pr=7458

```
======================================================================
FAIL: test_send_gemini_file_batches_s3_access_denied (cl.ai.tests.SendGeminiBatchesTest.test_send_gemini_file_batches_s3_access_denied)
Test error handling for S3 permission errors.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/local/lib/python3.13/unittest/case.py", line 651, in run
    self._callTestMethod(testMethod)
    
  File "/usr/local/lib/python3.13/unittest/case.py", line 606, in _callTestMethod
    if method() is not None:
    ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/unittest/mock.py", line 1432, in patched
    return func(*newargs, **newkeywargs)
    ^^^^^^^
  File "/usr/local/lib/python3.13/unittest/mock.py", line 1897, in _inner
    return f(*args, **kw)
    ^^^^^^^
  File "/opt/courtlistener/cl/ai/tests.py", line 1107, in test_send_gemini_file_batches_s3_access_denied
    self.assertIn("Access denied", str(context.exception))
    ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/unittest/case.py", line 1174, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
    ^^^^^^^^^^^
  File "/usr/local/lib/python3.13/unittest/case.py", line 732, in fail
    raise self.failureException(msg)
    ^^^^^^^^^^^^^^^
AssertionError: 'Access denied' not found in "Model 'gemini-2.5-pro' was shut down on 2026-06-17. See https://ai.google.dev/gemini-api/docs/deprecations"

----------------------------------------------------------------------
Ran 2190 tests in 263.861s
```

--- a/cl/ai/llm_providers/google.py
+++ b/cl/ai/llm_providers/google.py
@@ -21,11 +21,13 @@ DOWNLOADABLE_JOB_STATES = {
 # Maps model name to its shutdown date (None if no date announced).
 # See https://ai.google.dev/gemini-api/docs/deprecations
 SUPPORTED_GEMINI_MODELS: dict[str, date | None] = {
+    "gemini-3.5-flash": None,
     "gemini-3.1-pro-preview": None,
+    "gemini-3.1-flash-lite": date(2027, 5, 7),
     "gemini-3-flash-preview": None,
-    "gemini-2.5-pro": date(2026, 6, 17),
-    "gemini-2.5-flash": date(2026, 6, 17),
-    "gemini-2.5-flash-lite": date(2026, 7, 22),
+    "gemini-2.5-pro": date(2026, 10, 16),
+    "gemini-2.5-flash": date(2026, 10, 16),
+    "gemini-2.5-flash-lite": date(2026, 10, 16),
 }
 
 

--- a/cl/ai/tests.py
+++ b/cl/ai/tests.py
@@ -3,6 +3,7 @@ from datetime import date
 from unittest.mock import MagicMock, patch
 
 import pytest
+import time_machine
 from botocore.exceptions import ClientError
 from django.core.management import call_command
 from django.core.management.base import CommandError
@@ -90,12 +91,12 @@ class ValidateGeminiModelTest(SimpleTestCase):
     @patch("cl.ai.llm_providers.google.date")
     def test_model_past_shutdown_date_raises_error(self, mock_date):
         """Test that a model past its shutdown date raises ValueError."""
-        mock_date.today.return_value = date(2026, 6, 17)
+        mock_date.today.return_value = date(2026, 10, 16)
         mock_date.side_effect = lambda *args, **kw: date(*args, **kw)
         with self.assertRaises(ValueError) as context:
             GoogleGenAIBatchWrapper.validate_model("gemini-2.5-pro")
         self.assertIn("shut down", str(context.exception))
-        self.assertIn("2026-06-17", str(context.exception))
+        self.assertIn("2026-10-16", str(context.exception))
 
     def test_unsupported_model_raises_error(self):
         """Test that an unsupported model raises ValueError."""
@@ -705,6 +706,11 @@ class ResponseValidatorTest(SimpleTestCase):
         )
 
 
+# Freeze time so model-shutdown validation stays deterministic regardless of
+# the wall clock. The command builds GoogleGenAIBatchWrapper (which calls
+# validate_model) before reaching the S3/batch logic these tests exercise, so
+# a passed shutdown date would otherwise mask the behavior under test.
+@time_machine.travel("2026-01-01", tick=False)
 class SendGeminiBatchesTest(TestCase):
     """Tests for the send_gemini_file_batches management command."""
 

--- a/cl/ai/tests.py
+++ b/cl/ai/tests.py
@@ -81,22 +81,32 @@ class ValidateGeminiModelTest(SimpleTestCase):
         self.assertIn("does not have a shutdown date", warning_msg)
         self.assertIn("deprecations", warning_msg)
 
+    @patch.dict(
+        "cl.ai.llm_providers.google.SUPPORTED_GEMINI_MODELS",
+        {"test-model": date(2026, 1, 2)},
+    )
     @patch("cl.ai.llm_providers.google.date")
     def test_valid_model_before_shutdown_date(self, mock_date):
         """Test that a model before its shutdown date passes validation."""
+        # one day before shutdown
         mock_date.today.return_value = date(2026, 1, 1)
         mock_date.side_effect = lambda *args, **kw: date(*args, **kw)
-        GoogleGenAIBatchWrapper.validate_model("gemini-2.5-pro")
+        GoogleGenAIBatchWrapper.validate_model("test-model")
 
+    @patch.dict(
+        "cl.ai.llm_providers.google.SUPPORTED_GEMINI_MODELS",
+        {"test-model": date(2026, 1, 1)},
+    )
     @patch("cl.ai.llm_providers.google.date")
     def test_model_past_shutdown_date_raises_error(self, mock_date):
         """Test that a model past its shutdown date raises ValueError."""
-        mock_date.today.return_value = date(2026, 10, 16)
+        # one day past shutdown
+        mock_date.today.return_value = date(2026, 1, 2)
         mock_date.side_effect = lambda *args, **kw: date(*args, **kw)
         with self.assertRaises(ValueError) as context:
-            GoogleGenAIBatchWrapper.validate_model("gemini-2.5-pro")
+            GoogleGenAIBatchWrapper.validate_model("test-model")
         self.assertIn("shut down", str(context.exception))
-        self.assertIn("2026-10-16", str(context.exception))
+        self.assertIn("2026-01-01", str(context.exception))
 
     def test_unsupported_model_raises_error(self):
         """Test that an unsupported model raises ValueError."""

